### DISCUSSION TITLE
Is it possible to do configuration without CLI

### DISCUSSION THREAD

**Discussion by Crystalyyh** [great-expectations/great_expectations#5709] [Support] [answered]
Hi,
I have been doing testing for a while. Wondering if it is possible to create datasource, expectation suite, and checkpoint without using CLI.

**Comment by austiezr** [comment, marked as answer]
Hey @Crystalyyh ! Once you've initialized Great Expectations, you can create datasources, expectations & suites, and checkpoints programmatically. The CLI helps streamline this by provided notebooks w/ boilerplate code, but that code (or similar) can be used w/o invoking the CLI.

**Reply by Crystalyyh** [reply]
Thanks for resolving my problem, I sucessfully configured it =)
